lib/bootstrap.dart

```diff
@@ -0,0 +1,22 @@
+import 'dart:async';
+import 'dart:developer';
+
+import 'package:ax_dapp/app/app.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+Future<void> bootstrap(FutureOr<Widget> Function() bootstrapper) async {
+  FlutterError.onError = (details) {
+    log(details.exceptionAsString(), stackTrace: details.stack);
+  };
+
+  await runZonedGuarded(
+    () async {
+      await BlocOverrides.runZoned(
+        () async => runApp(await bootstrapper()),
+        blocObserver: AppBlocObserver(),
+      );
+    },
+    (error, stackTrace) => log(error.toString(), stackTrace: stackTrace),
+  );
+}
```

Comment:
Can you explain a bit what these are? I am assuming this has to do with some kind of scoping?

Reply:
https://medium.com/geekculture/mastering-flutter-zones-ea28fe6a6ca6 
Ahh I see this, first time I had heard of it. Looks very useful good work.

Reply:
It's like running the app in a safe container where you can intercept async errors, e.g.: from unhandled streams. It this case we can also run blocs zoned so that we can specify global configuration.